src/Client.js

```diff
@@ -1044,6 +1044,32 @@ class Client extends EventEmitter {
         return null;
     }
 
+    /**
+     * Gets an instance of a pinned message in a chat
+     * @param {string} chatId The chat ID
+     * @returns {Promise<Message>} Returns an instance of a pinned message in a chat
+     */
+    async getPinnedMessage(chatId) {
+        const pinnedMsg = await this.pupPage.evaluate(async (chatId) => {
+            const chatWid = window.Store.WidFactory.createWid(chatId);
+            const chat = await window.Store.Chat.find(chatWid);
+            if (!chat) return null;
+            await window.Store.Cmd.openChatAt(chat);
```

Comment:
Why we need to open the chat?
We have to find another solution 

We need to find a way to update the collection of pinned messages in another way

Reply:
@shirser121
> Why we need to open the chat?
 
The collection is populated only when you click on a pinned message banner inside the chat itself. Otherwise, the collection will remain empty.
So this is the only way to do so.

Open the chat in which you are looking for a pinned message → wait 500ms for the collection to be populated → get the pinned message

Reply:
We just need to find the inner function that did it, and call it directly

Reply:
@shirser121
> We just need to find the inner function that did it, and call it directly 

Unfortunately we don't have that function that would do the job.
I reversed the way it gets the pinned message and it happens after you enter the chat (here the collection is populated), and after you click on the banner it gets the message id from the collection. There is no function in the middle (at least I didn'tfind it)

Reply:
### This is a call stack:
![](https://github.com/pedroslopez/whatsapp-web.js/assets/93551621/7e1f3693-1838-4175-a43c-4b2e9060e978)

### This is a `j` function (right after the `onClick`):
![](https://github.com/pedroslopez/whatsapp-web.js/assets/93551621/1cc40fbf-595a-4fa1-a89a-898533bdc32d)

### It calls for an `M` function which appears as `I` in a call stack:
![](https://github.com/pedroslopez/whatsapp-web.js/assets/93551621/a09435ad-9b53-4ecb-866f-96ad4d819579)

### And here it is an `I` function that opens a chat at a message:
![](https://github.com/pedroslopez/whatsapp-web.js/assets/93551621/fb1e6ef1-91e0-4afd-ad74-d6e188faed37)

Reply:
@shirser121 
Maybe we have to look at a logic that happens immediately after you open a chat, when the collection is populated